File: 1941-check-if-all-characters-have-equal-number-of-occurrences/1941-check-if-all-characters-have-equal-number-of-occurrences.c
bool areOccurrencesEqual(char * s){
    int freq[26] = {0};
    int i = 0,val;
    
    while(s[i]){
        freq[s[i] - 'a']++;
        i++;
    }
    val = freq[s[0] - 'a'];
    for(i = 0; i < 26; i++){
        if(freq[i] && freq[i] != val)
            return false;
    }
    return true;
}

File: 1941-check-if-all-characters-have-equal-number-of-occurrences/1941-check-if-all-characters-have-equal-number-of-occurrences.swift
class Solution {
    func areOccurrencesEqual(_ s: String) -> Bool {
        if s.isEmpty { return false }
        if s.count == 1 { return true }
        
        var dictionary = [Character : Int]()
        
        for i in s {
            if dictionary[i] == nil {
                dictionary[i] = 1
            } else {
                dictionary[i]! += 1
            }
        }
        
        var firstValue = dictionary.first!.value
        
        for (key,value) in dictionary {
            if value != firstValue {
                return false
            }
        }
        
        return true
    }
}